### PR TITLE
Convert privatedns service to SDKv2

### LINKED ISSUE #3929: Update privatedns service to SDK v2

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Converts the privatedns service to azure-sdk-for-go version 2 and the `asyncpoller` framework.

**Which issue(s) this PR fixes**:

Fixes #3929

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [x] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

**Review by willie-yao — APPROVED** [review]
/lgtm

File: azure/services/privatedns/zone_client.go [hunk 1]
```diff
@@ -18,136 +18,103 @@ package privatedns
...
+
+	resp, err := azc.privatezones.Get(ctx, spec.ResourceGroupName(), spec.ResourceName(), nil)
+	if err != nil {
+		return nil, err
 	}
+	return resp.PrivateZone, nil
 }
 
 // CreateOrUpdateAsync creates or updates a private dns zone asynchronously.
 // It sends a PUT request to Azure and if accepted without error, the func will return a Future which can be used to track the ongoing
```
Comment (nojnhuh) [inline azure/services/privatedns/zone_client.go:62]: wow I found one:
```suggestion
// It sends a PUT request to Azure and if accepted without error, the func will return a poller which can be used to track the ongoing
```

**Review by nojnhuh — COMMENTED** [review]
/lgtm
/approve

/hold in case you're motivated enough to fix one extremely tiny ticky-tack thing.

File: azure/services/privatedns/zone_client.go (hunk 1 above)
Comment (mboersma) [reply azure/services/privatedns/zone_client.go]: Fixed, good eye!

**Pull request merged**